### PR TITLE
Switch to using Material instead of Multimaterial when there is only one submesh in glTF loaders

### PR DESCRIPTION
**Pull request by kcoley**

### CODE REVIEW

File: loaders/src/glTF/1.0/babylon.glTFLoader.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿/// <reference path="../../../../dist/preview release/babylon.d.ts"/>
+﻿/// <reference path="../../../../dist/preview release/babylon.d.ts"/>import { MultiMaterial } from "babylonjs";import { MultiMaterial } from "babylonjs";import { MultiMaterial } from "babylonjs";import { MultiMaterial } from "babylonjs";
```
Comment (sebavan) [inline loaders/src/glTF/1.0/babylon.glTFLoader.ts]: Replace by previous place.

File: loaders/src/glTF/2.0/babylon.glTFLoader.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-﻿/// <reference path="../../../../dist/preview release/babylon.d.ts"/>
+﻿/// <reference path="../../../../dist/preview release/babylon.d.ts"/>import { InstancingAttributeInfo } from "babylonjs";
```
Comment (sebavan) [inline loaders/src/glTF/2.0/babylon.glTFLoader.ts]: same

File: loaders/src/glTF/2.0/babylon.glTFLoader.ts [hunk 3]
```diff
@@ -1739,6 +1760,9 @@ module BABYLON.GLTF2 {
                         }
                     }
                 }
+                else if (mesh.material instanceof Material) {
+                    remaining = 1;
```
Comment (sebavan) [inline loaders/src/glTF/2.0/babylon.glTFLoader.ts]: ++

File: loaders/src/glTF/2.0/babylon.glTFLoader.ts [hunk 4]
```diff
@@ -1758,6 +1782,11 @@ module BABYLON.GLTF2 {
                         }
                     }
                 }
+                else if (mesh.material instanceof Material) {
+                    this._compileMaterialAsync(mesh.material, mesh, () => {
+                        onSuccess();
```
Comment (sebavan) [inline loaders/src/glTF/2.0/babylon.glTFLoader.ts]: if (--remaining === 0) {
                                     onSuccess();
                                 }

File: loaders/src/glTF/1.0/babylon.glTFLoader.ts [hunk 5]
```diff
@@ -690,14 +687,31 @@ module BABYLON.GLTF1 {
...
                 indexStarts.push(indexStarts.length === 0 ? 0 : indexStarts[indexStarts.length - 1] + indexCounts[indexCounts.length - 2]);
             }
         }
+        let material : Material | MultiMaterial;
+        if (subMaterials.length > 1) {
+            material = new MultiMaterial("multimat" + id, gltfRuntime.scene);
+            (material as MultiMaterial).subMaterials = subMaterials;
+        }
+        else {
+            material = new Material("multimat" + id, gltfRuntime.scene);
```
Comment (deltakosh) [inline loaders/src/glTF/1.0/babylon.glTFLoader.ts]: I would have picked a StandardMaterial here

File: loaders/src/glTF/2.0/babylon.glTFLoader.ts [hunk 6]
```diff
@@ -1739,6 +1760,9 @@ module BABYLON.GLTF2 {
                         }
                     }
                 }
+                else if (mesh.material instanceof Material) {
```
Comment (deltakosh) [inline loaders/src/glTF/2.0/babylon.glTFLoader.ts]: All materials have Material in their prototype chain so this test will always be true

**Pull request merged**